src/main/java/org/broadinstitute/consent/http/service/NihService.java

```diff
@@ -12,23 +13,31 @@
 import java.util.Calendar;
 import java.util.Date;
 import java.util.List;
+import java.util.Map;
 
 public class NihService {
 
     private ResearcherService researcherService;
+    private LibraryCardDAO libraryCardDAO;
 
     @Inject
-    public NihService(ResearcherService researcherService) {
+    public NihService(ResearcherService researcherService, LibraryCardDAO libraryCardDAO) {
         this.researcherService = researcherService;
+        this.libraryCardDAO = libraryCardDAO;
     }
 
-    public List<UserProperty> authenticateNih(NIHUserAccount nihAccount, AuthUser user) throws BadRequestException {
+    public List<UserProperty> authenticateNih(NIHUserAccount nihAccount, AuthUser authUser, Integer userId) throws BadRequestException {
         if (StringUtils.isNotEmpty(nihAccount.getNihUsername()) && !nihAccount.getNihUsername().isEmpty()) {
             nihAccount.setEraExpiration(generateEraExpirationDates());
             nihAccount.setStatus(true);
-            return researcherService.updateProperties(nihAccount.getNihMap(), user, false);
+            Map<String, String> userProp = researcherService.describeResearcherPropertiesMap(userId);
+            String eraCommonsId = (userProp == null) ? "" : userProp.getOrDefault(UserFields.ERA_COMMONS_ID.getValue(), "");
+            if (!eraCommonsId.equals("")) {
+              libraryCardDAO.updateEraCommonsForUser(userId, eraCommonsId);
```

Comment:
Quick question: `eraCommonsId` is the value saved in the UserProperties db right? But since the ResearcherProperties aren't being updated until the end of the function(line 38), is the libraryCard update saving what is essentially the "old" id value instead of the updated value on `nihAccount`?

Reply:
yeah you are right I just moved the update so it happens first and then the updated era commons id is used to update the library cards

Reply:
Have not reviewed this yet, but having two field names for the same value is going to be a nightmare to debug. We need to consolidate on one value somehow.

Reply:
We have a story that will remove this ambiguity (https://broadworkbench.atlassian.net/browse/DUOS-1095), but it's slated for SO functionality, not LC functionality. I think we can defer unifying the `eraCommonsId` and `nihUsername` fields until that story.